tools.go

```diff
@@ -1,8 +1,10 @@
-//go:build tools
-// +build tools
+//go:build generate
```

Comment:
I'd move the `go:generate` to the import to have it all in one place. If this is not desired, I can revert it, of course.

Reply:
If you now run go generate does it actually generate? Because this build will exclude it from the normal build process.

Reply:
According to https://pkg.go.dev/cmd/go/internal/generate:
```
The go generate tool also sets the build tag "generate" so that files may be examined by go generate but ignored during build.
```
It should generate it, yes. I tested it locally and it generated the doc. (that is also the reason, why a change in the doc is in this PR)

Reply:
Maybe this would be something to submit to the SDK sample projects too?